client/my-sites/plugins/plugins-list/index.jsx

```diff
@@ -0,0 +1,390 @@
+						plugin: pluginName,
+						numberOfSites: siteListSize
+					}
+				} );
+				break;
+			case 'n sites n plugins':
+				return this.translate( 'You are about to remove {{em}}%(numberOfPlugins)d plugins from %(numberOfSites)d sites{{/em}}.{{p}}This will deactivate the plugins and delete all associated files and data.{{/p}}', {
+					components: {
+						em: <em />,
+						p: <p />
+					},
+					args: {
+						numberOfPlugins: pluginsListSize,
+						numberOfSites: siteListSize
+					}
+				} );
+				break;
+		}
+	},
+
+	removePluginNotice() {
+		const message = (
+			<div>
+				<span>{ this.getConfirmationText() }</span>
+				<span>{ this.translate( 'Do you want to continue?' ) }</span>
+			</div>
+		);
+
+		acceptDialog(
+			message,
+			this.removeSelected,
+			this.translate( 'Remove', { context: 'Verb. Presented to user as a label for a button.' } )
+		);
+	},
+
+	removeSelected( accepted ) {
+		if ( accepted ) {
+			this.doActionOverSelected( 'removing', PluginsActions.removePlugin );
+			this.recordEvent( 'Clicked Remove Plugin(s)', true );
+		}
+	},
+
+	showDisconnectDialog() {
+		if ( this.state.disconnectJetpackDialog && ! this.props.notices.inProgress.length ) {
+			this.setState( { disconnectJetpackDialog: false } );
+			this.refs.dialog.open();
+		}
+	},
+
+	// Renders
+	render() {
+		const itemListClasses = classNames( 'list-cards-compact', 'plugins-list', {
+			'is-bulk-editing': this.state.bulkManagement
+		} );
+
+		if ( this.props.isPlaceholder ) {
+			return (
+				<span>
+					<SectionHeader key="plugins-list__section-placeholder" label={ this.props.header } className="plugins-list__section-actions is-placeholder" />
+					<div className={ itemListClasses }>{ this.renderPlaceholders() }</div>
+				</span>
+				)
+		}
+
+		if ( isEmpty( this.props.plugins ) ) {
+			return null;
+		}
+
+		return (
+			<span>
+				<PluginsListHeader label={ this.props.header }
+					isWpCom={ this.props.isWpCom }
+					isBulkManagementActive={ !! this.state.bulkManagement }
+					sites={ this.props.sites }
+					plugins={ this.props.plugins }
+					selected={ this.getSelected() }
+					toggleBulkManagement={ this.toggleBulkManagement }
+					updateAllPlugins={ this.updateAllPlugins }
+					pluginUpdateCount={ this.getPluginUpdateCount() }
+					activateSelected={ this.activateSelected }
+					deactiveAndDisconnectSelected={ this.deactiveAndDisconnectSelected }
+					deactivateSelected={ this.deactivateSelected }
+					setAutoupdateSelected={ this.setAutoupdateSelected }
+					unsetAutoupdateSelected={ this.unsetAutoupdateSelected }
+					removePluginNotice={ this.removePluginNotice }
+					setSelectionState={ this.setBulkSelectionState }
+					haveActiveSelected={ this.props.plugins.some( this.filterSelection.active.bind( this ) ) }
+					haveInactiveSelected={ this.props.plugins.some( this.filterSelection.inactive.bind( this ) ) } />
+				<div className={ itemListClasses }>{ this.renderPlugins() }</div>
+				{ this.props.isWpCom
```

Comment:
This could be written `! this.props.isWpCom && <DisconnectJetpackDialog ref="dialog" site={ this.props.site }...`